### PR TITLE
Refactoring docs to break into subpages

### LINKED ISSUE #119: Break README.md into document subpages?
Main `README.md` is getting pretty long. Should we break into separate pages?

+ Usage of the Analyzer API
+ Usage of the Query Builder API
+ ...

@Chriskamphuis @PepijnBoers @x65han thoughts?

### PR DESCRIPTION
**Pull request by lintool**
Closes #119 - also adopts @rodrigonogueira4 's suggestion of what to focus documentation on.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -112,267 +112,48 @@ for i in range(0, 10):
...
-index_reader = index.IndexReader('indexes/index-robust04-20191213/')
+doc = searcher.doc('LA071090-0047')
 ```
 
-Use `terms()` to grab an iterator over all terms in the collection, i.e., the dictionary.
-Note that these terms are _analyzed_.
-Here, we only print out the first 10:
+From `doc`, you can access its `contents` as well as its `raw` representation.
+The `contents` hold the representation of what's actually indexed; the `raw` representation is usually the original "raw document".
+A simple example can illustrate this distinction: for an article from CORD-19, `raw` holds the complete JSON of the article, which obviously includes the artcle contents, but has metadata and other information as well.
```
Comment (PepijnBoers) [inline README.md:124]: artcle -> article

File: docs/usage-collection.md [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+collection = collection.Collection('HtmlCollection', 'collections/cacm/')
+generator = index.Generator('DefaultLuceneDocumentGenerator')
+
+for (i, fs) in enumerate(collection):
+    for (j, doc) in enumerate(fs):
+        parsed = generator.create_document(doc)
+        docid = parsed.get('id')            # FIELD_ID
+        raw = parsed.get('raw')             # FIELD_RAW
+        contents = parsed.get('contents')   # FIELD_BODY
+        print('{} {} -> {} {}...'.format(i, j, docid, contents.strip().replace('\n', ' ')[:50]))
```
Comment (PepijnBoers) [inline docs/usage-collection.md:27]: Use f-strings for formatting?

File: README.md (hunk 1 above)
Comment (lintool) [reply README.md]: good catch!

File: docs/usage-collection.md (hunk 2 above)
Comment (lintool) [reply docs/usage-collection.md:27]: I tried, but according to: https://www.python.org/dev/peps/pep-0498/#escape-sequences

> Backslashes may not appear anywhere within expressions.

Comment (PepijnBoers) [reply docs/usage-collection.md:27]: Ah okay I didn't know that!

**Pull request merged**